src/System.Collections/tests/Generic/Queue/QueueTests.cs

```diff
@@ -31,6 +32,95 @@ public void Ctors()
         }
 
         [Theory]
+        [MemberData("Collections")]
+        public void Ctor_IEnumerable(IEnumerable<int> collection)
+        {
+            var q = new Queue<int>(collection);
+
+            Assert.Equal(collection, q);
+
+            foreach (int item in collection)
+            {
+                Assert.Equal(item, q.Dequeue());
+            }
+        }
+
+        [Theory]
+        [MemberData("Collections")]
+        public void Ctor_IEnumerable_EnqueueDequeue(IEnumerable<int> collection)
+        {
+            var q = new Queue<int>(collection);
+            q.Enqueue(int.MaxValue);
+            q.Enqueue(int.MinValue);
+
+            foreach (int item in collection)
+            {
+                Assert.Equal(item, q.Dequeue());
+            }
+            Assert.Equal(int.MaxValue, q.Dequeue());
+            Assert.Equal(int.MinValue, q.Dequeue());
+        }
+
+        [Theory]
+        [MemberData("Collections")]
+        public void Ctor_IEnumerable_Dequeue(IEnumerable<int> collection)
+        {
+            var q = new Queue<int>(collection);
+            IEnumerator<int> e = collection.GetEnumerator();
+
+            e.MoveNext();
+            q.Dequeue();
+
+            while (e.MoveNext())
+            {
+                Assert.Equal(e.Current, q.Dequeue());
+            }
+        }
+
+        [Theory]
+        [MemberData("Collections")]
+        public void Ctor_IEnumerable_DequeueEnqueue(IEnumerable<int> collection)
+        {
+            var q = new Queue<int>(collection);
+            IEnumerator<int> e = collection.GetEnumerator();
+
+            e.MoveNext();
+            q.Dequeue();
+
+            q.Enqueue(int.MaxValue);
+
+            while (e.MoveNext())
+            {
+                Assert.Equal(e.Current, q.Dequeue());
+            }
+            Assert.Equal(int.MaxValue, q.Dequeue());
+        }
+
+        public static IEnumerable<object[]> Collections
+        {
+            get
+            {
+                foreach (int size in new int[] { 1, 2, 3, 4, 5 })
```

Comment:
Maybe a large binary-round and a large non-binary-round size would be worth adding to this set. Also zero to catch any future regression that breaks on empty collections.